File: bindings/dotnet/examples/Free.cs
using System;
using Hyperic.Sigar;

public class Free {

    public static void Main() {
        Sigar sigar = new Sigar();

	Mem mem = sigar.Mem();
	Swap swap = sigar.Swap();

        System.Console.WriteLine("\tTotal\tUsed\tFree");

        System.Console.WriteLine("Mem:\t" +
                                 mem.Total / 1024 + "\t" +
                                 mem.Used / 1024 + "\t" +
                                 mem.Free / 1024);

        System.Console.WriteLine("Swap:\t" +
                                 swap.Total / 1024 + "\t" +
                                 swap.Used / 1024 + "\t" +
                                 swap.Free / 1024);

        System.Console.WriteLine("RAM:\t" + mem.Ram + "MB");
    }
}

File: bindings/perl/examples/free.pl
#!/usr/bin/perl

use strict;
use Sigar;

my $sigar = new Sigar;

my $mem = $sigar->mem;

my $swap = $sigar->swap;

print "\tTotal\tUsed\tFree\n";

print "Mem:    " .
      ($mem->total / 1024) . "\t" .
      ($mem->used / 1024) . "\t" .
      ($mem->free / 1024) . "\n";

print "Swap:   " .
      ($swap->total / 1024) . "\t" .
      ($swap->used / 1024) . "\t" .
      ($swap->free / 1024) . "\n";

print "RAM:    " . $mem->ram . "MB\n";
